src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -2302,6 +2335,31 @@ internal override int MetadataArity
                     return (object)containingType == null ? 0 : containingType.MetadataArity;
                 }
             }
+
+            public override bool IsNativeInt { get; }
+
+            internal override NamedTypeSymbol AsNativeInt()
+            {
+                Debug.Assert(this.SpecialType == SpecialType.System_IntPtr || this.SpecialType == SpecialType.System_UIntPtr);
+
+                return IsNativeInt ? this : new PENamedTypeSymbolNonGeneric(this);
+            }
+
+            internal override bool Equals(TypeSymbol t2, TypeCompareKind comparison, IReadOnlyDictionary<TypeParameterSymbol, bool> isValueTypeOverrideOpt = null)
+            {
+                if ((object)t2 == null) return false;
+                if ((object)t2 == this) return true;
+
+                var other = t2 as PENamedTypeSymbolNonGeneric;
+                if ((object)other != null &&
+                    (this.IsNativeInt || other.IsNativeInt) &&
+                    this.SpecialType == other.SpecialType)
```

Comment:
Is the intent here for this to match two `IntPtr` symbols where one is being treated as `nint` and one is not?

Reply:
Yes. I've added a comment to replace or update this.

---
In reply to: [352874882](https://github.com/dotnet/roslyn/pull/39157#discussion_r352874882) [](ancestors = 352874882)